### DISCUSSION TITLE
Tried to read FOO from a place where one of its dependencies were overridden but the provider is not.

### DISCUSSION THREAD

**Discussion by derolf** [rrousselGit/riverpod#2522] [Q&A]
I added 

```
             final dependencyElement =
                  readProviderElement<Object?>(dependency);
             final targetDependencyElement = targetElement.container
                  .readProviderElement<Object?>(dependency);
```

and

```
Expected container: ${dependencyElement._container._overrideForProvider} ${dependencyElement._container.hashCode}
Actual container:   ${targetDependencyElement._container._overrideForProvider} ${targetDependencyElement._container.hashCode}
```

before the assertion to debug a bit deeper:

```
════════ Exception caught by widgets library ═══════════════════════════════════
Tried to read mapLayersProvider:MapLayersProvider#194be(null) from a place where one of its dependencies were overridden but the provider is not.

Expected container: {} 3137226806
Actual container:   {} 3137226806

To fix this error, you can add "dependencies" to FOO such that we have:
```

So, the assertion triggers, meaning `dependencyElement != targetDependencyElement`. The containers are IDENTICAL, but I have two elements for the same provider.

What could be the reason?

**Comment by rrousselGit** [comment]
Who knows. You haven't given me the code triggering the problem; only how you tried to modify the error message.

**Reply by derolf** [reply]
Yeah, I know. Just trying to figure out whether overriden providers are the ONLY possible cause or if I am screwing it up in a different t way…

**Reply by derolf** [reply]
The root cause was a combination of an exception thrown in a build function and a leaked ref that was used by an async function. That somehow resulted in having two elements for the same provider.

**Reply by rrousselGit** [reply]
Ah I see. I guess the Element was destroyed but you kept a reference on it. Interesting

**Reply by derolf** [reply]
Yes, and the error message was "misleading"...
